### PR TITLE
Allow to use reference to content

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Allows to use URL reference for K8S/OS components. 
 - `local` field renamed to `reference` and allows to use absolute of devfile-relative paths;
 - `localContent` renamed to `referenceContent` without changes 

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/12733

#### Release Notes
N/A

#### Docs PR
Included in PR

### CODE REVIEW

File: wsmaster/che-core-api-devfile/README.md [hunk 1]
```diff
@@ -101,28 +101,27 @@ List of available Che plugins and more information about registry can be found o
...
-Contents of the `local` file is currently read _ONLY_ if the devfile and local file both placed in the same public GitHub repository. 
-So, alternatively, if you need to post devfile with such components to REST API, contents of K8S/Openshift list can be embedded into devfile using `localContent` field:
+Alternatively, if you need to post devfile with such components to REST API, contents of K8S/Openshift list can be embedded into devfile using `referenceContent` field:
 
 ```
 ...
   components:
     - name: mysql
       type: kubernetes
       local: petclinic.yaml
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/README.md]: Should it be renamed to `reference?`

File: wsmaster/che-core-api-devfile/README.md [hunk 2]
```diff
@@ -101,28 +101,27 @@ List of available Che plugins and more information about registry can be found o
...
-       localContent: |
+       referenceContent: |
```
Comment (sleshchenko) [inline wsmaster/che-core-api-devfile/README.md]: Please, put it on the same level as `name`, `type`, `reference` are

**Review by sleshchenko — APPROVED** [review]
LGTM

**Comment by riuvshin** [comment]
ci-build

File: wsmaster/che-core-api-devfile/README.md (hunk 1 above)
Comment (mshaposhnik) [reply wsmaster/che-core-api-devfile/README.md]: fixed

File: wsmaster/che-core-api-devfile/README.md (hunk 2 above)
Comment (mshaposhnik) [reply wsmaster/che-core-api-devfile/README.md]: fixed

**Comment by mshaposhnik** [comment]
ci-build

**Comment by mshaposhnik** [comment]
ci-build

**Comment by mshaposhnik** [comment]
ci-build

**Comment by mshaposhnik** [comment]
ci-build

**Comment by mshaposhnik** [comment]
ci-build

**Comment by mshaposhnik** [comment]
ci-build

**Comment by mshaposhnik** [comment]
ci-build

**Comment by mshaposhnik** [comment]
ci-test

**Comment by artaleks9** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1675/) doesn't show any regression against this Pull Request.

**Review by skabashnyuk — APPROVED** [review]

**Pull request merged**